backend/fms_core/template_importer/row_handlers/normalization/normalization.py

```diff
@@ -0,0 +1,91 @@
+from fms_core.template_importer.row_handlers._generic import GenericRowHandler
+
+from fms_core.models import ProcessMeasurement
+
+from fms_core.services.container import get_container, get_or_create_container
+from fms_core.services.sample import get_sample_from_container, transfer_sample, update_sample
+from fms_core.services.property_value import create_process_measurement_properties
+
+from fms_core.utils import check_truth_like, convert_concentration_from_nm_to_ngbyul
+
+
+class NormalizationRowHandler(GenericRowHandler):
+
+    def process_row_inner(self, source_sample, destination_sample, process_measurement, process_measurement_properties):
+        source_sample_obj, self.errors['sample'], self.warnings['sample'] = get_sample_from_container(
+            barcode=source_sample['container']['barcode'],
+            coordinates=source_sample['coordinates'])
+
+        destination_container_dict = destination_sample['container']
+
+        parent_barcode = destination_container_dict['parent_barcode']
+        if parent_barcode:
+            container_parent, self.errors['parent_container'], self.warnings['parent_container'] = get_container(
+                barcode=parent_barcode)
+        else:
+            container_parent = None
+
+        if source_sample and ((parent_barcode and container_parent) or not parent_barcode):
```

Comment:
This is an overly complicated test it seem to be sufficient to test source_sample and (container_parent or not parent_barcode) in principle to get the same result and a simple logic.